test/jwt_filter_claims_plug_test.exs

```diff
@@ -0,0 +1,115 @@
+    @the_plug Jwt.Plugs.FilterClaims
+    @opts []
+
+    test "Missing claims returns 401" do
+        conn = conn(:get, "/protected")
+
+        conn = @the_plug.call(conn, @the_plug.init(@opts))
+
+        assert401(conn)
+    end
+
+    test "Missing or empty options returns 401" do
+        conn = build_conn_with_claims()
+
+        conn = @the_plug.call(conn, @the_plug.init([]))
+
+        assert401(conn)
+    end
+
+    test "Filter passes single exact email address" do
+        conn = build_conn_with_claims()
+
+        conn = @the_plug.call(conn, [%{"email" => "^alejandro.mezcua@byteabyte.net$"}])
+
+        assertFilterPassed(conn)
+    end
+
+    test "Filter rejects single non matching exact email address" do
+        conn = build_conn_with_claims()
+
+        conn = @the_plug.call(conn, [%{"email" => "^not_in_claims@byteabyte.net$"}])
+
+        assert401(conn)
+    end
+
+    test "Filter passes any email address from domain" do
+        conn = build_conn_with_claims()
+
+        conn = @the_plug.call(conn, [%{"email" => "^[_A-Za-z0-9-\\+]+(\\.[_A-Za-z0-9-]+)*@byteabyte.net$"}])
+
+        assertFilterPassed(conn)
+    end
+
+    test "Filter passes domain property" do
+        conn = build_conn_with_claims()
+
+        conn = @the_plug.call(conn, [%{"hd" => "^byteabyte.net$"}])
+
+        assertFilterPassed(conn)
+    end
+
+    test "Filter passes when email in token is last in config list" do
+      conn = build_conn_with_claims()
+
+      conn = @the_plug.call(conn, [ %{"email" => ["^someone@byteabyte.net$", "^alejandro.mezcua@byteabyte.net$"]} ])
+
+      assertFilterPassed(conn)
+    end
+
+    test "Filter fails when email in token is not in config list" do
+      conn = build_conn_with_claims()
+
+      conn = @the_plug.call(conn, [ %{"email" => ["^someone@byteabyte.net$", "^other@byteabyte.net$"]} ])
+
+      assert401(conn)
+    end
+
+    test "Filter passes on two different valid properties" do
+      conn = build_conn_with_claims()
+
+      conn = @the_plug.call(conn, [ %{"iss" => "^https://accounts.google.com$"},
+                                    %{"email" => "^alejandro.mezcua@byteabyte.net$"}])
+
+      assertFilterPassed(conn)
+    end
+
+    test "Filter fails on two different properties, one invalid" do
+      conn = build_conn_with_claims()
+
+      conn = @the_plug.call(conn, [ %{"iss" => "^https://accounts.google.com$"},
+                                    %{"email" => "^someone@byteabyte.net$"}])
+
+      assert401(conn)
+    end
+
+    test "Filter fails if requested claim is not present" do
+        conn = build_conn_with_claims()
+
+        conn = @the_plug.call(conn, [ %{"notpresent" => "^anyvalue$"} ])
```

Comment:
There is no whitespace around parentheses/brackets most of the time, but here there is.